contracts/root/withdrawManager/WithdrawManager.sol

```diff
@@ -144,7 +156,44 @@ contract WithdrawManager is WithdrawManagerStorage /* , IWithdrawManager */ {
     emit ExitStarted(_exitObject.owner, priority, _exitObject.token, _exitObject.receiptAmountOrNFTId);
   }
 
-  function deleteExit(uint256 exitId) external isProofValidator {
+  function challengeExit(uint256 exitId, uint256 inputId, bytes calldata challengeData)
+    external
+  {
+    PlasmaExit storage exit = exits[exitId];
+    Input storage input = exit.inputs[inputId];
+    require(
+      exit.token != address(0x0) && input.signer != address(0x0),
+      "Invalid exit or input id"
+    );
+    // IPredicate(exit.predicate).verifyDeprecation(exit, inputId, data, registry);
+    bool isChallengeValid = IPredicate(exit.predicate).verifyDeprecation(
```

Comment:
Basically, now we are relying on predicate owner to code for proof validation.

Reply:
Yes, this is assuming that we'll whitelist a predict in the core plasma.